### PR TITLE
add ssl feature extractor

### PR DESCRIPTION
**Pull request by DongjiGao**
The SSL extractor is supported by [S3PRL](https://github.com/s3prl/s3prl)

### CODE REVIEW

File: lhotse/features/ssl.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        ), f"S3PRL dose not suport model: {self.config.ssl_model}."
+
+        ssl_model = getattr(hub, self.config.ssl_model)()
+        self.ssl_model = ssl_model.to(self.config.device)
+
+    def feature_dim(self, sampling_rate: int) -> int:
+        assert (
+            sampling_rate == 16000
+        ), f"All the upstream models in S3PRL now only support 16 kHz audio."
+        feature_dim = 1024
```
Comment (pzelasko) [inline lhotse/features/ssl.py:55]: You might want to move feature_dim, sampling_rate, and frame_shift to the config to let people override it for different models (even if today they all use the same setting)

**Review by pzelasko — COMMENTED** [review]
Looks great, thank you! I left one comment, otherwise LGTM

**Comment by csukuangfj** [comment]
Shall we add tests for it?

**Comment by pzelasko** [comment]
I'm afraid that it will require downloading a big pretrained model from somewhere, which won't be very convenient either for CI or local testing. WDYT?

**Comment by csukuangfj** [comment]
In icefall we upload the pretrained models to huggingface and download them in CI for testing using git lfs.

A file an be several hundred MB and it's very fast to download in the CI.

**Comment by pzelasko** [comment]
Cool! @DongjiGao could you contribute a test for it? You can use a decorator on the test function to make the test optional. Then add `s3prl.hub` installation in `.github/workflows/unit_tests.yml` to enable the test being run.

Decorator example:

```
@pytest.mark.skipif(not is_module_available("s3prl.hub", reason="The test requires s3prl to run.")
def test_s3prl_feature_extractor():
    ...
```

**Comment by DongjiGao** [comment]
> Cool! @DongjiGao could you contribute a test for it? You can use a decorator on the test function to make the test optional. Then add `s3prl.hub` installation in `.github/workflows/unit_tests.yml` to enable the test being run.
> 
> Decorator example:
> 
> ```
> @pytest.mark.skipif(not is_module_available("s3prl.hub"), reason="The test requires s3prl to run.")
> def test_s3prl_feature_extractor():
>     ...
> ```

Sure.

**Comment by desh2608** [comment]
@DongjiGao I'm wondering if perhaps you could add a comment somewhere (maybe as a docstring for the class), with a link to where users could find other pretrained models (and their configs like frame_shift, feature_dim, etc.)?

File: test/features/test_s3prl.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+from lhotse.utils import compute_num_frames, compute_num_samples, is_module_available
+
+
+@pytest.fixture()
+def recording():
+    return Recording.from_file("test/fixtures/libri/libri-1088-134315-0000.wav")
+
+
+@pytest.mark.skipif(
+    not is_module_available("s3prl.hub"), reason="The test requires s3prl to run."
```
Comment (pzelasko) [inline test/features/test_s3prl.py:14]:
```suggestion
    not is_module_available("s3prl"), reason="The test requires s3prl to run."
```

File: test/features/test_s3prl.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+)
+def test_s3prl_feature_extractor_default_config(recording):
+    feature_extractor = S3PRLSSL()
+    y = feature_extractor.extract(recording.load_audio(), recording.sampling_rate)
+    assert np.shape(y) == (802, 1024)
+
+
+@pytest.mark.skipif(
+    not is_module_available("s3prl.hub"), reason="The test requires s3prl to run."
```
Comment (pzelasko) [inline test/features/test_s3prl.py:23]:
```suggestion
    not is_module_available("s3prl"), reason="The test requires s3prl to run."
```

**Review by pzelasko — COMMENTED** [review]
You'll need to make these changes to let the tests pass

**Comment by pzelasko** [comment]
Also please remember about adding s3prl installation to the CI ` .github/workflows/unit_tests.yml`

**Comment by DongjiGao** [comment]
> @DongjiGao I'm wondering if perhaps you could add a comment somewhere (maybe as a docstring for the class), with a link to where users could find other pretrained models (and their configs like frame_shift, feature_dim, etc.)?

Adding it now.

**Comment by DongjiGao** [comment]
> Also please remember about adding s3prl installation to the CI ` .github/workflows/unit_tests.yml`

Sorry I was using a personal token previously and was blocked to change that file. I have added the s3prl installation now.

**Comment by DongjiGao** [comment]
@pzelasko @csukuangfj @desh2608 Do you think the code is good to check in?

**Review by desh2608 — APPROVED** [review]

**Comment by desh2608** [comment]
LGTM.

**Pull request merged**